evm_loader/cli/src/rpc/db.rs

```diff
@@ -0,0 +1,110 @@
+use solana_sdk::{
+    account::Account,
+    pubkey::Pubkey,
+};
+use tokio_postgres::{ connect, Error, Client};
+use postgres::{ NoTls};
+use serde::{Serialize, Deserialize };
+use tokio::task::block_in_place;
+use std::convert::TryFrom;
+
+
+#[derive(Serialize, Deserialize, Debug, PartialEq, Eq, Clone)]
+pub struct DBConfig{
+    pub host: String,
+    pub port: String,
+    pub database: String,
+    pub user: String,
+    pub password: String,
+}
+
+#[derive(Debug)]
+pub struct PostgresClient {
+    pub slot: u64,
+    client: Client,
+}
+
+pub fn block<F, Fu, R>(f: F) -> R
+    where
+        F: FnOnce() -> Fu,
+        Fu: std::future::Future<Output = R>,
+{
+    block_in_place(|| {
+        let handle = tokio::runtime::Handle::current();
+        handle.block_on(f())
+    })
+}
+
+impl PostgresClient {
+    pub fn new(config: &DBConfig, slot: u64) -> Self {
+        let connection_str= format!("host={} port={} dbname={} user={} password={}",
+                                    config.host, config.port, config.database, config.user, config.password);
+
+        let (client, connection) = block(|| async {
+            connect(&connection_str, NoTls).await
+        }).unwrap();
```

Comment:
Please handle it properly without unwrap.
You should also try to reconnect several times, not just one, and after that display an error message if something is wrong.

Reply:
fixed

Reply:
fixed